starknet-foundry/crates/cast/src/starknet_commands/account/create.rs

```diff
@@ -0,0 +1,187 @@
+    let mut items: serde_json::Value =
+        serde_json::from_str(&contents).expect("failed to parse json file");
+
+    let network_value = get_network(network)?.get_value();
+
+    if !items[network_value][&name].is_null() {
+        return Err(anyhow!(
+            "Account with provided name already exists in this network"
+        ));
+    }
+
+    items[network_value][&name] = json!({
+        "private_key": format!("{:#x}", private_key.secret_scalar()),
+        "public_key": format!("{:#x}", public_key.scalar()),
+        "address": format!("{address:#x}"),
+        "salt": format!("{salt:#x}"),
+        "deployed": false,
+    });
+
+    std::fs::write(
+        accounts_file_path.clone(),
+        serde_json::to_string_pretty(&items).unwrap(),
+    )?;
+
+    let mut output = vec![(
+        "message",
+        format!("Account successfully created. Prefund generated address with at least {max_fee} tokens. \
+         It is good to send more in the case of higher demand, max_fee * 2 = {}", max_fee * 2),
+    ), ("address", format!("{address:#x}"))];
+    if add_profile {
+        output.push((
+            "add_profile",
+            add_created_profile_to_configuration(name, network.to_string(), url),
+        ));
+    }
+
+    Ok(output)
+}
+
+pub fn print_account_create_result(
+    account_create_result: Result<Vec<(&'static str, String)>>,
+    int_format: bool,
+    json: bool,
+) -> Result<()> {
+    match account_create_result {
+        Ok(mut values) => {
+            values.insert(0, ("command", "Create account".to_string()));
+            print_formatted(values, int_format, json, false)?;
+        }
+        Err(error) => {
+            print_formatted(vec![("error", error.to_string())], int_format, json, true)?;
+        }
+    };
+
+    Ok(())
+}
+
+pub fn add_created_profile_to_configuration(name: String, network: String, url: String) -> String {
+    let manifest_path = find_manifest_path(None).expect("Failed to obtain Scarb.toml file path");
+
+    let metadata = scarb_metadata::MetadataCommand::new()
+        .inherit_stderr()
+        .manifest_path(&manifest_path)
+        .no_deps()
+        .exec()
+        .context(
+            "Failed to read Scarb.toml manifest file, not found in current nor parent directories",
+        )
+        .unwrap();
+
+    let package = &metadata.packages[0].manifest_metadata.tool;
+
+    let property = get_property(package, &Some(name.clone()), "account");
+    if property.is_ok() {
+        return format!("Failed to add {name} profile to the Scarb.toml. Profile already exists");
+    }
+
+    let toml_string = {
+        let mut tool_sncast = toml::value::Table::new();
+        let mut new_profile = toml::value::Table::new();
+
+        new_profile.insert("network".to_string(), Value::String(network));
+        new_profile.insert("url".to_string(), Value::String(url));
+        new_profile.insert("account".to_string(), Value::String(name.clone()));
```

Comment:
Would it be possible to use `CastConfig` here?

Reply:
Maybe, but I do not see benefits. We would have to create CastConfig instance and then insert each value to the `new_profile`, right?

Reply:
It probably depends on `toml` package - perhaps it can do something like `CastConfig.to_string()`? I don't know, that was my question :D

Reply:
Unfortunately it does not work as expected :/
![image](https://github.com/foundry-rs/starknet-foundry/assets/61014013/d99f2b49-54e9-4b63-857a-d308720159a8)

Reply:
https://github.com/orgs/foundry-rs/projects/3/views/2?filterQuery=milestone:Cast&pane=issue&itemId=34135288